app/models/concerns/waste_carriers_engine/can_have_registration_attributes.rb

```diff
@@ -86,12 +86,27 @@ module CanHaveRegistrationAttributes
       field :total_fee,                                        type: String
 
       scope :search_term, lambda { |term|
-        escaped_term = Regexp.escape(term) if term.present?
+        if term.present?
```

Comment:
Ignore what I said previously about the `return` clause! I do think this block would benefit from being moved to a different method – `process_phone_number_term` or something like that.